### PR TITLE
Get tokens from GNU make if possible

### PR DESCRIPTION
**Pull request by nobu**

### CODE REVIEW

File: lib/mspec/commands/mspec.rb [hunk 1]
```diff
@@ -93,6 +93,20 @@ def multi_exec(argv)
...
+        w = IO.for_fd($2.to_i(10), "wb", autoclose: false)
+      rescue
+        r.close if r
+        nil
+      else
+        jobserver = [r, w]
+        jobtokens = r.read_nonblock(1024)
+        cores = jobtokens.size
+      end
+    end
```
Comment (eregon) [inline lib/mspec/commands/mspec.rb]: Is there an advantage from getting it from `make` flags over getting it via e.g. `/proc/cpuinfo`?

Comment (nobu) [inline lib/mspec/commands/mspec.rb]: You can manage how many processors to be used consistently.

Comment (eregon) [inline lib/mspec/commands/mspec.rb]: This seems a fair amount of code specific to one build system, I would prefer if it was in [MRI MSpec config](https://github.com/ruby/ruby/blob/trunk/spec/default.mspec) or runner and exported as an environment variable like `JOBS`.
Do you think that would be possible?

**Comment by eregon** [comment]
@nobu I'm thinking to add a `JOBS` (or is there a better/more standard name?) environment variable to customize the number of parallel workers.

**Comment by nobu** [comment]
Monkey-pached at ruby/ruby@18f4cf13b5dbdd3c72067b3717763646bd61b5ce

**Pull request closed without merging**